distributed/utils_test.py

```diff
@@ -1199,13 +1225,17 @@ def popen(args, **kwargs):
             _terminate_process(proc)
         finally:
             # XXX Also dump stdout if return code != 0 ?
-            out, err = proc.communicate()
-            if dump_stdout:
-                print("\n\nPrint from stderr\n  %s\n=================\n" % args[0][0])
-                print(err.decode())
+            if flush_output:
+                out, err = flush_future.result()
+                ex.shutdown()
+            else:
+                out, err = proc.communicate()
+            assert not err
 
-                print("\n\nPrint from stdout\n=================\n")
-                print(out.decode())
+            if dump_stdout:
+                print("\n" + "-" * 27 + " Subprocess stdout/stderr" + "-" * 27)
```

Comment:
```suggestion
                print("\n" + "-" * 27 + " Subprocess stdout" + "-" * 27)
```
Are you not printing `err` because `kwargs["stderr"] = subprocess.STDOUT` means it's already been printed?

Reply:
Yes, they are merged together because I was afraid of the subprocess getting stuck on stdout while a test is reading from stderr and vice versa.

Reply:
Oh I see yes, stderr is redirected into stdout, which is then captured.